extensions/amp-story-shopping/0.1/test/test-amp-story-shopping-config.js

```diff
@@ -231,5 +205,89 @@ describes.realWin(
+      await createAmpStoryShoppingConfig(null, invalidConfig);
+      expect(spy).to.have.been.calledWith(
+        'AMP-STORY-SHOPPING-CONFIG',
+        errorString
+      );
+    });
+
+    it('should fail config validation because an expected string JSON value is not a valid HTML id', async () => {
+      const invalidConfig = JSON.parse(JSON.stringify(defaultInlineConfig));
+      const invalidValue = '1234city-pop'; // productId starts with a number, so it is an invalid HTML Id
+      invalidConfig['items'][0]['productId'] = invalidValue;
+
+      const errorString = `Error: productId ${invalidValue} is not a valid HTML Id`;
+      const spy = env.sandbox.spy(user(), 'warn');
+      await createAmpStoryShoppingConfig(null, invalidConfig);
+      expect(spy).to.have.been.calledWith(
+        'AMP-STORY-SHOPPING-CONFIG',
+        errorString
+      );
+    });
+
+    it('should fail config validation because an expected number JSON value is not a valid number', async () => {
+      const invalidConfig = JSON.parse(JSON.stringify(defaultInlineConfig));
+      const invalidValue = 'two dozen watermelons'; // two dozen watermelons is not an actual price.
+      invalidConfig['items'][0]['productPrice'] = invalidValue;
+
+      const errorString = `Error: Value ${invalidValue} for field productPrice is not a number`;
+      const spy = env.sandbox.spy(user(), 'warn');
+      await createAmpStoryShoppingConfig(null, invalidConfig);
+      expect(spy).to.have.been.calledWith(
+        'AMP-STORY-SHOPPING-CONFIG',
+        errorString
+      );
+    });
+
+    it('should fail config validation because an expected string JSON value is not a valid currency code symbol', async () => {
+      const invalidConfig = JSON.parse(JSON.stringify(defaultInlineConfig));
+      const invalidValue = 'ZABAN'; // This is not a valid currency symbol code
+      invalidConfig['items'][0]['productPriceCurrency'] = invalidValue;
+
+      const errorString = `Error: productPriceCurrency ${invalidValue} is not a valid currency code`;
+      const spy = env.sandbox.spy(user(), 'warn');
+      await createAmpStoryShoppingConfig(null, invalidConfig);
+      expect(spy).to.have.been.calledWith(
+        'AMP-STORY-SHOPPING-CONFIG',
+        errorString
+      );
+    });
+
+    it('should fail config validation because an expected string JSON value is not a valid url', async () => {
+      expectAsyncConsoleError(errorString, 1);
+      const invalidConfig = JSON.parse(JSON.stringify(defaultInlineConfig));
+      const invalidValue = 'http://zapp'; // This is not a valid url
+      invalidConfig['items'][0]['productUrl'] = invalidValue;
+
+      const errorString = `Error: productUrl ${invalidValue} is not a valid URL. (Error: amp-story-shopping-config productImages source must start with "https://" or "//" or be relative and served from either https or from localhost. Invalid value: ${invalidConfig['items'][0]['productUrl'][0]})`;
```

Comment:
You can just replace `${invalidConfig['items'][0]['productUrl'][0]}` with `${invalidValue}`, especially since the 'productUrl' value is supposed to be a string. The `[0]` part will currently result in the first character being used (i.e., `"Invalid value: h"` instead of `"Invalid value: http://zapp"`)